be/src/storage/lake/tablet_manager.h

```diff
@@ -141,7 +141,7 @@ class TabletManager {
     // only for TEST purpose
     void TEST_set_global_schema_cache(int64_t index_id, TabletSchemaPtr schema);
 
-    void update_segment_cache_size(std::string_view key);
+    void update_segment_cache_size(std::string_view key, const void* segment_hint = nullptr);
```

Comment:
why not use `const Segment* segment_hint = nullptr`

Reply:
I am thinking of change it to `intptr_t`, because here we don't want to do any member accessing to the type, just an memory address or the object address that matters. typeinfo can be erased as much as possible.